ecommerce/extensions/payment/processors/paystack.py

```diff
@@ -0,0 +1,200 @@
+
+    def get_return_url(self):
+        redirect_url = reverse('paystack:execute')
+        ecommerce_base_url = get_ecommerce_url()
+        return '{}{}'.format(ecommerce_base_url, redirect_url)
+
+    def get_basket_amount(self, basket):
+        return str((basket.total_incl_tax * 100).to_integral_value())
+
+    def get_paystack_custom_fields_data(self, basket):
+        """
+        Return Paystack Custom fields.
+        """
+        product = basket.all_lines()[0].product
+        return [
+            {
+                'display_name': 'Order Number',
+                'variable_name': 'order_id',
+                'value': basket.order_number
+            },
+            {
+                'display_name': 'Course Id',
+                'variable_name': 'course_id',
+                'value': product.course_id
+            },
+            {
+                'display_name': 'Course Name',
+                'variable_name': 'course_title',
+                'value': product.course.name
+            }
+        ]
+
+    def get_transaction_parameters(self, basket, request=None, use_client_side_checkout=True, **kwargs):
+        """
+        Get Payment form URL from paystack.
+
+        Arguments:
+            basket (Basket): The basket of products being purchased.
+            request (Request, optional): A Request object which is used to construct PayPal's `return_url`.
+            use_client_side_checkout (bool, optional): This value is not used.
+            **kwargs: Additional parameters; not used by this method.
+
+        Returns:
+            dict: paystack-specific parameters required to complete a transaction. Must contain a URL
+                of payment form to which users can be directed in order to approve a newly created payment.
+
+        Raises:
+            GatewayError: Indicates a general error or unexpected behavior on the part of PayPal which prevented
+            a payment from being created.
+        """
+        return_url = self.get_return_url()
+        data = {
+            'amount': self.get_basket_amount(basket),
+            'email': basket.owner.email,
+            'callback_url': return_url,
+            'metadata': {
+                'cancel_action:': self.cancel_url,
+                'order_number': basket.order_number,
+                'basket_id': basket.id,
+                'custom_fields': self.get_paystack_custom_fields_data(basket)
+            },
+        }
+
+        success, response = self.paystack_client.handler(INITIALIZE_TRANSACTION_CODE, data)
+
+        if success:
+            logger.info("Successfully got hosted Paystack payment page for basket [%d].", basket.id)
+            data = response.get('data')
+            if data:
+                return {'payment_page_url': data.get('authorization_url')}
+
+        logger.error("Failed to get Paystack payment form for basket [%d].", basket.id)
+        raise GatewayError('Paystack payment creation failure: unable to get Paystack form token')
```

Comment:
```suggestion
        raise GatewayError("Paystack payment creation failure: unable to get Paystack form token")
```
As you are using double quotes around text strings everywhere so please follow this to make your code consistent.